src/nsolid/nsolid_bpf.cc

```diff
@@ -1,9 +1,78 @@
 #include "nsolid_bpf.h"
 #include "nsolid_api.h"
 
+// Check if libbpf headers are available
+#if __has_include(<bpf/bpf.h>) && __has_include(<bpf/libbpf.h>)
+#define HAVE_LIBBPF 1
+#include <bpf/bpf.h>
+#include <bpf/libbpf.h>
+#else
+#define HAVE_LIBBPF 0
+#endif
+
+#include <string>
+#include <vector>
+#include <mutex>
+
 namespace node {
 namespace nsolid {
 
+#if HAVE_LIBBPF
+std::vector<struct ::bpf_object*> EbpfLoader::loaded_objects_;
+static std::mutex bpf_objects_mutex;
+#endif
```

Comment:
Why don't make the mutex part of the `EbpfLoader` class: we should only be accessing loaded_objects_ via `EbpfLoader` methods.